Index: modules/python/CMakeLists.txt
===================================================================
--- modules/python/CMakeLists.txt	(revision 3119)
+++ modules/python/CMakeLists.txt	(working copy)
@@ -57,5 +57,9 @@
             ARCHIVE DESTINATION "Python${PYTHON_VERSION_MAJOR_MINOR}/Lib/site-packages" COMPONENT main
             )
 else()
-    install(FILES ${LIBRARY_OUTPUT_PATH}/cv${CVPY_SUFFIX} DESTINATION ${PYTHON_PACKAGES_PATH})
+    #install(FILES ${LIBRARY_OUTPUT_PATH}/cv${CVPY_SUFFIX} DESTINATION ${PYTHON_PACKAGES_PATH})
+    install(TARGETS ${the_target} 
+            RUNTIME DESTINATION ${PYTHON_PACKAGES_PATH} COMPONENT main
+            LIBRARY DESTINATION ${PYTHON_PACKAGES_PATH} COMPONENT main
+            ARCHIVE DESTINATION ${PYTHON_PACKAGES_PATH} COMPONENT main)
 endif()
Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 3119)
+++ CMakeLists.txt	(working copy)
@@ -24,6 +24,9 @@
     endif()
 endif()
 
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
 set(CMAKE_CONFIGURATION_TYPES "Debug;Release" CACHE STRING "Configs" FORCE)
 set(CMAKE_C_FLAGS_MINSIZEREL "" CACHE INTERNAL "" FORCE)
 set(CMAKE_C_FLAGS_RELWITHDEBINFO "" CACHE INTERNAL "" FORCE)
